src/v3/verifier.ts

```diff
@@ -61,62 +50,124 @@ interface InternalVerifierOptions {
-      if (this.config.consumerVersionSelectors) {
-        config.consumerVersionSelectorsString = this.config.consumerVersionSelectors.map(
-          (s) => JSON.stringify(s)
-        );
-      }
+  private validateConfiguration() {
+    const config: VerifierV3Options & InternalVerifierOptions = {
+      ...this.config,
+    };
 
-      if (!this.config.provider) {
-        reject(new ConfigurationError('Provider name is required'));
-      }
-      if (
-        (isEmpty(this.config.pactUrls) || !this.config.pactUrls) &&
-        !this.config.pactBrokerUrl
-      ) {
-        reject(
-          new ConfigurationError(
-            'Either a list of pactUrls or a pactBrokerUrl must be provided'
-          )
+    if (this.config.logLevel && !isEmpty(this.config.logLevel)) {
+      setLogLevel(this.config.logLevel);
+    }
+
+    if (this.config.validateSSL === undefined) {
+      this.config.validateSSL = true;
+    }
+
+    if (this.config.changeOrigin === undefined) {
+      this.config.changeOrigin = false;
+
+      if (!this.isLocalVerification()) {
+        this.config.changeOrigin = true;
+        logger.debug(
+          `non-local provider address ${this.config.providerBaseUrl} detected, setting 'changeOrigin' to 'true'. This property can be overridden.`
         );
       }
+    }
 
-      try {
-        PactNative.verify_provider(this.config, (err, val) => {
-          if (err || !val) {
-            logger.debug('In verify_provider callback: FAILED with', err, val);
-            reject(err);
-          } else {
-            logger.debug('In verify_provider callback: SUCCEEDED with', val);
-            resolve(val);
-          }
-        });
-        logger.debug('Submitted test to verify_provider');
-      } catch (e) {
-        reject(e);
-      }
-    });
+    if (isEmpty(this.config)) {
+      throw new ConfigurationError('no configuration provided to verifier');
+    }
+
+    // This is just too messy to do on the rust side. neon-serde would have helped, but appears unmaintained
+    // and is currently incompatible
+    if (this.config.consumerVersionSelectors) {
+      config.consumerVersionSelectorsString = this.config.consumerVersionSelectors.map(
+        (s) => JSON.stringify(s)
+      );
+    }
+
+    if (!this.config.provider) {
+      throw new ConfigurationError('provider name is required');
+    }
+
+    if (
+      (isEmpty(this.config.pactUrls) || !this.config.pactUrls) &&
+      !this.config.pactBrokerUrl
+    ) {
+      throw new ConfigurationError(
+        'a list of pactUrls or a pactBrokerUrl must be provided'
+      );
+    }
+
+    return config;
+  }
+
+  // Run the Verification CLI process
+  private runProviderVerification() {
+    return (server: http.Server) =>
+      new Promise((resolve, reject) => {
+        const opts = {
+          providerStatesSetupUrl: `${this.address}:${server.address().port}${
+            this.stateSetupPath
+          }`,
+          ...omit(this.config, 'handlers'),
```

Comment:
You can do this in ES6 without lodash:

```
{ handlers, ...rest } = this.config;
```

Reply:
That's doing something slightly different (omitting `handlers` altogether), but in this case it actually doesn't matter. I think omit can go entirely, because the rust verifier won't do anything with that prop.